protostar/commands/test/environments/setup_execution_environment.py

```diff
@@ -0,0 +1,53 @@
+from typing import List
+
+from starkware.starknet.business_logic.execution.objects import CallInfo
+
+from protostar.commands.test.cheatcodes import (
+    DeclareCheatcode,
+    DeployCheatcode,
+    DeployContractCheatcode,
+    PrepareCheatcode,
+)
+from protostar.commands.test.environments.execution_environment import (
+    ExecutionEnvironment,
+)
+from protostar.commands.test.starkware.test_execution_state import TestExecutionState
+from protostar.starknet.cheatcode import Cheatcode
+from protostar.starknet.cheatcode_factory import CheatcodeFactory
+from protostar.utils.data_transformer_facade import DataTransformerFacade
+
+
+class SetupExecutionEnvironment(ExecutionEnvironment):
+    def _cheatcode_factory(self) -> CheatcodeFactory:
+        return SetupCheatcodeFactory(self.state)
+
+
+class SetupCheatcodeFactory(CheatcodeFactory):
+    def __init__(self, state: TestExecutionState):
+        self._state = state
+
+    def build(
+        self,
+        syscall_dependencies: Cheatcode.SyscallDependencies,
+        internal_calls: List[CallInfo],
+    ) -> List[Cheatcode]:
+        data_transformer = DataTransformerFacade(self._state.starknet_compiler)
+
+        declare_cheatcode = DeclareCheatcode(
+            syscall_dependencies,
+            disable_hint_validation=self._state.disable_hint_validation_in_external_contracts,
+            cairo_path=self._state.include_paths,
+        )
+        prepare_cheatcode = PrepareCheatcode(syscall_dependencies, data_transformer)
+        deploy_cheatcode = DeployCheatcode(syscall_dependencies, internal_calls)
+        return [
```

Comment:
Shouldn't `mock_call`, `prank`, and other should be available here?

Reply:
I think all except `expect_revert`, `expect_events` i think? @kasperski95

Reply:
> I think all except expect_revert, expect_events i think?

Did that

Reply:
OK, please folks, review current state and resolve this convo if that's correct :)

Reply:
done